modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -0,0 +1,610 @@
+                    ptrs[i] += i_k*steps[i][k];
+                idx = next_idx;
+            }
+            for (int i2 = 0; i2 < n2; i2++)
+            {
+                const T* ptr1 = (const T*)(ptrs[1] + steps[1][ndims-2]*i2);
+                const T* ptr2 = (const T*)(ptrs[second] + steps[second][ndims-2]*i2);
+                T* ptr = (T*)(ptrs[0] + steps[0][ndims-2]*i2);
+                if (ninputs <= 2) {
+                    if (dp1 == 1 && dp2 == 1) {
+                        for (int i1 = 0; i1 < n1; i1++)
+                            ptr[i1] = saturate_cast<T>(f(ptr1[i1], ptr2[i1])*scale);
+                    } else {
+                        for(int i1 = 0; i1 < n1; i1++, ptr1 += dp1, ptr2 += dp2, ptr += dp)
+                            *ptr = saturate_cast<T>(f(*ptr1, *ptr2)*scale);
+                    }
+                } else {
+                    for (int i1 = 0; i1 < n1; i1 += di1, ptr += di1) {
+                        di1 = BLOCK_SIZE < n1-i1 ? BLOCK_SIZE : n1-i1;
+                        if (dp1 == 1 && dp2 == 1) {
+                            for (int j = 0; j < di1; j++)
+                                blck[j] = f(ptr1[j], ptr2[j]);
+                            ptr1 += di1;
+                            ptr2 += di1;
+                        } else {
+                            for(int j = 0; j < di1; j++, ptr1 += dp1, ptr2 += dp2)
+                                blck[j] = f(*ptr1, *ptr2);
+                        }
+                        for(i = 2; i < ninputs; i++) {
+                            int dp_i = steps[i+1][ndims-1]/sizeof(T);
+                            const T* ptr_i = (const T*)(ptrs[i+1] +
+                                    steps[i+1][ndims-2]*i2) + i1*dp_i;
+                            if (dp_i == 1) {
+                                if (i < ninputs-1) {
+                                    for (int j = 0; j < di1; j++)
+                                        blck[j] = f(blck[j], ptr_i[j]);
+                                } else {
+                                    for (int j = 0; j < di1; j++)
+                                        ptr[j] = saturate_cast<T>(f(blck[j], ptr_i[j]) * scale);
+                                }
+                            } else {
+                                if (i < ninputs-1) {
+                                    for (int j = 0; j < di1; j++, ptr_i += dp_i)
+                                        blck[j] = f(blck[j], *ptr_i);
+                                } else {
+                                    for (int j = 0; j < di1; j++, ptr_i += dp_i)
+                                        ptr[j] = saturate_cast<T>(f(blck[j], *ptr_i) * scale);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // TODO: AutoBuffer for all allocations, this should be called and cached inside finailize()
```

Comment:
TODO: cache everything inside finalize(), test the performance

Reply:
I think we keep buf in binary_forward/nary_forward. Tested with vector and finalize(), performances are almost the same.